server/datastore/mysql/mdm.go

```diff
@@ -229,25 +229,26 @@ FROM (
 		if prof.Platform == "windows" {
 			winProfUUIDs = append(winProfUUIDs, prof.ProfileUUID)
 		} else {
-			if strings.HasPrefix(prof.ProfileUUID, "x") {
+			if strings.HasPrefix(prof.ProfileUUID, fleet.MDMAppleDeclarationUUIDPrefix) {
 				macDeclUUIDs = append(macDeclUUIDs, prof.ProfileUUID)
 				continue
 			}
 
 			macProfUUIDs = append(macProfUUIDs, prof.ProfileUUID)
 		}
 	}
-	labels, err := ds.listProfileLabelsForProfiles(ctx, winProfUUIDs, macProfUUIDs)
+	labels, err := ds.listProfileLabelsForProfiles(ctx, winProfUUIDs, macProfUUIDs, macDeclUUIDs)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	declLabels, err := ds.listDeclarationLabelsForDeclarations(ctx, macDeclUUIDs)
-	if err != nil {
-		return nil, nil, err
-	}
+	// // TODO: do we want to load labels separately for declarations?
```

Comment:
I ended up removing this separate function because it wasn't necessary with the platform-agnostic approach, but we could handle it either way. I commented it out for now in case we want to add it back.

Reply:
I think this makes sense! should we delete the old code?